### ISSUE TITLE
Allow specifying custom action buttons for nodes

### ISSUE THREAD

**Issue by andy1547** [helion3/inspire-tree-dom#29]
This is just a placeholder for a potential enhancement (will design later).

It would be great if the configuration allowed for custom action buttons supplementing or replacing the basic add/edit/remove functionality. 
The ability to specify these on a node level would be most flexible benefiting ACL environments where each node may have custom functionality associated with it.

I haven't yet thought of a good way of doing this which doesn't cause the inspire-tree project to leak to the inspire-tree-dom project, am I right in thinking the inspire-tree project can be used standalone as a tree data structure?

**Comment by viveleroi** [comment]
Yes, the core repo is meant to be a pure API primarily for users wanting to do custom view layers.

My first thought would be to allow users to use the configuration object to customize buttons - they can provide a class, a click handler, etc. If configured, that would override the default buttons. There would still need to be a way to include the defaults.

Adding configuration params is the least "leaky" I can imagine. The tree core doesn't do anything with them - they just get passed through to the DOM.
